### ISSUE TITLE
Question Visibility into available ndarray methods?

### ISSUE THREAD

**Issue by javadba** [ramonhagenaars/nptyping#3]
Is it possible for the type hints to show the ndarray methods such as shape, argmax/argmin etc?

**Comment by ramonhagenaars** [comment]
Hi javadba. Seems like an improvement that is very legitimate and useful!

**Comment by javadba** [comment]
Can you provide some quick hint/guidance - then I'll take a stab?

**Comment by ramonhagenaars** [comment]
I've done a shot (see [pull request](https://github.com/ramonhagenaars/nptyping/pull/4/files)). Could you have a look at it? It lacks the right signatures of the methods though; need to do that still. 

If you have a better idea of doing it, please let me know! 🙂

**Comment by javadba** [comment]
if you can figure out how to do _one_ of the method signatures I can do the
grunt work to replicate that pattern to the other methods. I'm pretty
motivated since numpy is and will be my life for the foreseeable future.

Am Di., 4. Juni 2019 um 12:39 Uhr schrieb Ramon Hagenaars <
notifications@github.com>:

**Comment by ramonhagenaars** [comment]
I think we are limited to the method parameters and their default values as numpy does not have typehints.

So for example for the ``flatten`` method, we would get:
```python
@functools.wraps(np.ndarray.flatten)
def flatten(self, order='C'): ...
```

I wouldn't write this manually but create a little script to generate it for you. Something like this (I haven't tested it!):

```python
import inspect
import numpy as np
import Array  # import the Array from nptyping

for elem in dir(np.array([42])):  # loop through all attributes of np.ndarray
    if elem not in dir(Array):  # only use the attributes that are not already in Array
        if is_method:  # TODO: use inspect to see if something is a method
            signature = np.inspect(getattr(np.ndarray, elem)
            print('@functools.wraps(...')  # use functools.wraps to copy the documentation and stuff
            print(f'def ...etc')  # print the result to stdout so you can copy the code
            print()
        else:
            print(elem)  # attributes that are not methods can simply be copied
            print()
```

**Comment by ramonhagenaars** [comment]
This issue has been resolved in [0.3.0](https://pypi.org/project/nptyping/0.3.0/).
